### PR TITLE
Adapt i-set commands to confirm to storage requirements

### LINKED ISSUE #106: Integrate i-set commands with ingredients' storage
so that the feature of setting ingredients' levels can still work with the storage in place

### PR DESCRIPTION
**Pull request by Diwu-Yi**
Adapt i-set commands and relevant tests to confirm to storage requirements

### CODE REVIEW

**Review by Persdre — DISMISSED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/SetCommand.java [hunk 1]
```diff
@@ -53,15 +53,32 @@ public CommandResult execute(Model model) throws CommandException {
...
-        Ingredient updatedIngredient = createSetIngredient(target, setIngredientDescriptor);
+        Ingredient ingredientToEdit = null;
+        boolean isNoChange = false;
 
-        if (!ingredientToEdit.isSameIngredient(updatedIngredient)
-                && model.hasIngredient(updatedIngredient)) {
+        for (Ingredient ingredient : lastShownList) {
+            if (ingredient.getIngredientName().equals(targetName)) {
+                ingredientToEdit = ingredient;
+                if (ingredientToEdit.getAmount().equals(setIngredientDescriptor.getAmount().get())) {
```
Comment (yanbingtao) [inline src/main/java/seedu/address/logic/commands/SetCommand.java]: You may want to change this part to if (AAA && BBB) to avoid the deep nesting? Thanks

Comment (Diwu-Yi) [reply src/main/java/seedu/address/logic/commands/SetCommand.java:62]: good point, changing now

Comment (Diwu-Yi) [reply src/main/java/seedu/address/logic/commands/SetCommand.java]: noted, editing now

Comment (yanbingtao) [reply src/main/java/seedu/address/logic/commands/SetCommand.java]: 👍

**Comment by Diwu-Yi** [comment]
#105

**Review by yanbingtao — APPROVED** [review]
LGTM

**Review by Persdre — APPROVED** [review]
LGTM

**Pull request merged**